stacker/actions/diff.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _normalize_json(self, json_str):
+        """Takes a string representing a JSON object and normalizes it"""
+        obj = json.loads(json_str)
+        json_str = json.dumps(obj, sort_keys=True, indent=4)
+        lines = json_str.split("\n")
+        result = []
+        for line in lines:
+            result.append(line + "\n")
+        return result
+
+    def _diff_stack(self, stack, **kwargs):
+        """Handles the diffing a stack in CloudFormation vs our config"""
+        if not build.should_submit(stack) or not build.should_update(stack):
+            return SKIPPED
+
+        try:
+            old_stack = self.provider.get_template(stack.fqn)
+        except exceptions.StackDoesNotExist:
+            old_stack = None
+
+        new_stack = self._normalize_json(stack.blueprint.rendered)
+        logger.info("============== Stack: %s ==============", stack)
+        if not old_stack:
+            logger.info("New template contents:")
+            sys.stdout.write(''.join(new_stack))
```

Comment:
I decided to follow the diff command's lead of showing the new file/contents.  I briefly considered a --verbose option, but decided it wasn't worth the effort.